src/bootstrap.coffee

```diff
@@ -111,7 +111,10 @@ bootstrapper.startBootstrapping = ->
 		readConfigAndEnsureUUID()
 		.tap ->
 			loadPreloadedApps()
-		.tap ->
-			bootstrapOrRetry()
+		.tap (uuid) ->
+			if bootstrapper.offlineMode
+				return knex('config').insert({ key: 'uuid', value: uuid })
```

Comment:
Before the my change the following was happening:

`bootstrapper.startBootstrapping` is called every time the supervisor starts. Then it will call `readConfig` and also a `knex('config').select('value').where(key: 'uuid')`. If it finds the uuid in sqlite, `startBoostrapping` just resolves to its value. Otherwise, it calls `readConfigAndEnsureUUID`, then `loadPreloadedApps` and then `bootstrapOrRetry`.

The problem is that `bootstrapOrRetry` is responsible for eventually setting the uuid value in sqlite, but because it's in offline mode this never happens. And that means that every time the supervisor starts in offline mode it ends up in the never resolving `bootstrapOrRetry` and running `loadPreloadedApps` again and again.